sync3/extensions/account_data.go

```diff
@@ -21,6 +22,8 @@ func (r *AccountDataRequest) Name() string {
 type AccountDataResponse struct {
 	Global []json.RawMessage            `json:"global,omitempty"`
 	Rooms  map[string][]json.RawMessage `json:"rooms,omitempty"`
+	// which rooms have had account data loaded from the DB in this response
+	loadedRooms map[string]bool
```

Comment:
```suggestion
	loadedRooms map[string]struct{}
```
Maybe?

Reply:
It's easier to read if they are bools, and we don't need the space optimisation here, so I err'd on clarity over performance.